src/app/clusters/ota-requestor/DefaultOTARequestor.cpp

```diff
@@ -671,16 +671,20 @@ void DefaultOTARequestor::RecordNewUpdateState(OTAUpdateStateEnum newState, OTAC
     if ((newState == OTAUpdateStateEnum::kIdle) && (mCurrentUpdateState != OTAUpdateStateEnum::kIdle))
     {
         IdleStateReason idleStateReason = MapErrorToIdleStateReason(error);
-
-        // Inform the driver that the core logic has entered the Idle state
+        // Update the new state before handling the state transition
+        mCurrentUpdateState = newState;
```

Comment:
Worried about updating `mCurrentUpdateState` in multiple places could be error prone (in case there are future scenarios that get added and we forget to update the state). Maybe could do something like:

```
OTAUpdateStateEnum prevState = mCurrentUpdateState;
mCurrentUpdateState = newState;
if ((prevState != OTAUpdateStateEnum::kIdle) && (newState == OTAUpdateStateEnum::kIdle))
    {
        IdleStateReason idleStateReason = MapErrorToIdleStateReason(error);
        mOtaRequestorDriver->HandleIdleStateEnter(idleStateReason);
    }
    else if ((prevState == OTAUpdateStateEnum::kIdle) && (newState != OTAUpdateStateEnum::kIdle))
    {
        mOtaRequestorDriver->HandleIdleStateExit();
    }
```

Reply:
Done.